sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -178,13 +195,94 @@ internal class ViewAnnotationManagerImpl(
 
     if (viewAnnotationOptions.isEmpty()) return null
     val coordinates = coordinatesFromAnnotations(viewAnnotationOptions)
-    val paddings = calculateEdgeInsets(viewAnnotationOptions, edgeInsets)
-    return mapboxMap.cameraForCoordinates(
+    var cameraOptionForCoordinates = mapboxMap.cameraForCoordinates(
       coordinates,
-      paddings,
+      EdgeInsets(0.0, 0.0, 0.0, 0.0),
       bearing,
       pitch
     )
+
+    var isCorrectBound = false
+    // keep north, east, west and southmost Annotation and its frame to adjust paddings for zoom.
+    var north: Pair<ViewAnnotationOptions, Rect?>? = null
+    var east: Pair<ViewAnnotationOptions, Rect?>? = null
+    var west: Pair<ViewAnnotationOptions, Rect?>? = null
+    var south: Pair<ViewAnnotationOptions, Rect?>? = null
+
+    while (!isCorrectBound) {
+      val zoom = cameraOptionForCoordinates.zoom
+      isCorrectBound = true
+      viewAnnotationOptions.forEach { options ->
+        val frame = getViewAnnotationOptionsFrame(options)
+        val annotationBounds = calculateCoordinateBoundForAnnotation(options, zoom)
+        if (north == null || calculateCoordinateBoundForAnnotation(north!!.first, zoom).north() < annotationBounds.north()) {
```

Comment:
maybe get rid of `!!` similar to how you do it in L275? we try to avoid `!!` in SDK code

Reply:
I think here its fine, because we are already checking for null in the first option of `or` logic. if its null, then it won't execute next option of `or` logic. 
although i have added another check below for nullability. (that code shouldn't trigger in normal case because we have already checking for all the nullability conditions before, but good to include extra check).